lightning/src/offers/invoice.rs

```diff
@@ -220,6 +220,16 @@ pub struct InvoiceWithDerivedSigningPubkeyBuilder<'a> {
 	signing_pubkey_strategy: DerivedSigningPubkey,
 }
 
+/// A variant of [`InvoiceBuilder`] that indicates how the signing public key was set.
+///
+/// This is not exported to bindings users as builder patterns don't map outside of move semantics.
+pub enum InvoiceBuilderVariant<'a> {
+	/// An [`InvoiceBuilder`] that uses a derived signing public key.
+	Derived(InvoiceBuilder<'a, DerivedSigningPubkey>),
+	/// An [`InvoiceBuilder`] that uses an explicitly set signing public key.
+	Explicit(InvoiceBuilder<'a, ExplicitSigningPubkey>),
+}
```

Comment:
I'm not sure yet, but we may instead want to try making different `VerifiedInvoiceRequest` types(w/ and w/o derived keys). Then we'd have have two versions of `create_invoice_builder_from_invoice_request` for each type.